### PR TITLE
[upgrade] Add system state wrapper object

### PR DESCRIPTION
**Pull request by lxfind**
This PR introduces a wrapper object for the system state. The wrapper object has ID 0x5, and all access to the system state object is done indirectly through this wrapper object. Note that since this is already a lot of refactoring, I am keeping the wrapper object simple in this PR: it simply wraps the system state. In future PRs we will make the system state object a dynamic field object of the wrapper.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [x] breaking change for a client SDKs
- [x] breaking change for FNs (FN binary must upgrade)
- [x] breaking change for validators or node operators (must upgrade binaries)
- [x] breaking change for on-chain data layout
- [x] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-benchmark/tests/simtest.rs
Comment (emmazzz) [inline crates/sui-benchmark/tests/simtest.rs:69]: What's this change for?

File: crates/sui-types/src/sui_system_state.rs
Comment (emmazzz) [inline crates/sui-types/src/sui_system_state.rs:385]: In your next PR which will make the actual system state a dynamic field of the wrapper, I suppose you will be loading the dynamic field here? Do we have use cases where we may want to load not just the current version of the system state, but also the previous versions? If so this function should probably take in a version number too.

File: crates/sui-framework/sources/governance/sui_system.move
Comment (emmazzz) [inline crates/sui-framework/sources/governance/sui_system.move:82]: Calling this "wrapper" might be confusing to users of this module, who shouldn't need to know about the wrapping we are doing here. What if we call the wrapper "SuiSystemState" and the actual content "SuiSystemStateInner"? No strong preference though.

File: crates/sui-benchmark/tests/simtest.rs
Comment (lxfind) [reply crates/sui-benchmark/tests/simtest.rs:69]: I had a bug earlier where this call would run forever. Adding a timeout just so that it's easier to debug similar issues.

File: crates/sui-framework/sources/governance/sui_system.move
Comment (lxfind) [reply crates/sui-framework/sources/governance/sui_system.move:82]: Good question. I am not sure. 
My consideration is that dapps (explorer, wallet and etc) should be getting SuiSystemState for clarity.

File: crates/sui-types/src/sui_system_state.rs
Comment (lxfind) [reply crates/sui-types/src/sui_system_state.rs:385]: I haven't seen a need for that, but will keep it in mind

File: crates/sui-framework/sources/governance/sui_system.move
Comment (lxfind) [reply crates/sui-framework/sources/governance/sui_system.move:82]: Let me think about that more and maybe rename it in a separate PR because that would probably lead to a whole lot of changes everywhere.

File: crates/sui-framework/sources/governance/sui_system.move
Comment (amnn) [inline crates/sui-framework/sources/governance/sui_system.move:86]: Any reason to make it a dynamic *object* field (vs just a dynamic field) -- i.e. will we need to know the ID of the inner object off-chain?  Asking because there is an ongoing conversation about getting rid of dynamic object field, in favour of indexing wrapped objects.

File: crates/sui-framework/sources/governance/sui_system.move
Comment (amnn) [reply crates/sui-framework/sources/governance/sui_system.move:83]: +1 to @emmazzz's naming suggestion, as per the offline discussion!

Comment (amnn) [reply crates/sui-framework/sources/governance/sui_system.move:83]: FWIW, I think the size of this PR will go down quite a bit if you do the rename proactively, rather than in a separate PR -- no strong compulsion on my side to do it that way though.

**Review by amnn — COMMENTED** [review]
Change LGTM, but I'll leave it to folks who are closer to this work to do the final stamp.

Would encourage doing the rename to `SuiSysteStateInner` and `SuiSystemState` proactively -- the PR will become much smaller.

File: crates/sui-framework/sources/governance/sui_system.move
Comment (amnn) [reply crates/sui-framework/sources/governance/sui_system.move:86]: cc @tnowacki

File: crates/sui-benchmark/tests/simtest.rs
Comment (mystenmark) [reply crates/sui-benchmark/tests/simtest.rs:69]: can you move the timeout inside of `test_simulated_load`, so that it respects the duration passed in (instead of a hard coded timeout), and covers all tests?

**Review by mystenmark — COMMENTED** [review]
One small comment, otherwise looks good to me.

File: crates/sui-framework/sources/governance/sui_system.move
Comment (lxfind) [reply crates/sui-framework/sources/governance/sui_system.move:83]: hmm are you suggesting that we make Rust data structure name different from Move names?

File: crates/sui-framework/sources/governance/sui_system.move
Comment (lxfind) [reply crates/sui-framework/sources/governance/sui_system.move:86]: We don't need to know the ID of the inner object off-chain, as long as there is an easy way to access this child object in Rust

**Review by emmazzz — APPROVED** [review]
approving to unblock

File: crates/sui-framework/sources/governance/sui_system.move
Comment (amnn) [reply crates/sui-framework/sources/governance/sui_system.move:86]: I think you will, in any case, need to access the inner object via its parent, so doing it via a dynamic field will save you one step.

**Pull request merged**
